reducers/fileReducer.js

```diff
@@ -0,0 +1,19 @@
+import { SET_FILE_DATA } from '../actions';
+
+const initialState = {
+  fileData: [],
+};
+
+const fileReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SET_FILE_DATA:
+      return {
+        ...state,
+        fileData: action.payload,
```

Comment:
It seems only CSV file is working for now.  Some possible solutions: 
1. convert GeoJSON, Shapfile, and CSV data content into a uniform internal data structure that will be fed into other components
2. write a util function to deal with different data content before feed into other components

Again, here if we use Typescript, we need to define the data type of `action.payload`, the UI component will get the data type defs and raise warnings when we consume the data passed in, therefore helping us to get a better design. 

Let me know what you think. Thanks!